### PR TITLE
rfc: specify cli version api

### PR DESCRIPTION
**Pull request by msanft**
### Proposed change(s)

- Add specification for the CLI version compatibility API
  - The API should be exposed via the [version API](https://github.com/edgelesssys/constellation/tree/main/internal/versionsapi) and provide information on compatibility between a specific CLI version of a Constellation version and Kubernetes versions.

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: rfc/cli-api.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+The build pipeline produces lookup tables for compatibility information that are uploaded to S3 and can be accessed via HTTP.
+The lookup tables are organized in a directory structure that allows to look up the compatibility for a given Constellation version.
+
+The following HTTP endpoints are available:
+
+- `GET /constellation/v1/ref/<REF>/stream/<STREAM>/<VERSION>/cli/`
+  - [`info.json` returns the CLI compatibility lookup table for the given Constellation version.](#cli-lookup-table)
+
+## CLI lookup table
```
Comment (katexochen) [inline rfc/cli-api.md:15]: The term "lookup table" is somewhat ambiguous in our API specs and daily usage (I know the image-api rfc still uses it). I would prefer something like "info artifact" to differentiate this from the lookup table design we use in the (tighter) versions API.

File: rfc/cli-api.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+```
+/constellation/v1/ref/<REF>/stream/<STREAM>/<VERSION>/cli/info.json
+```
+
+```json
+{
+  "version": "<VERSION>",
+  "ref": "<REF>",
+  "stream": "<STREAM>",
+  "cli": "v1.2.3",
```
Comment (katexochen) [inline rfc/cli-api.md:28]: how does the "cli" field differ from the "version" field?

Comment (msanft) [reply rfc/cli-api.md:28]: Do we always build a corresponding CLI version to a Constellation release?

Comment (katexochen) [reply rfc/cli-api.md:28]: This object only exists if a CLI exists. You know that there is a CLI with this versions as you used the versions API to request a CLI version (cli.json) and got the path that consists of ref/stream/version. So you know that this CLI exists. The existing information about existence is stored in the versions API.

File: rfc/cli-api.md (hunk 1 above)
Comment (derpsteb) [reply rfc/cli-api.md:15]: another idea: "cli info".

**Comment by derpsteb** [comment]
Seconding katexochen's comments, lgtm apart from that.

**Review by katexochen — APPROVED** [review]
lgtm

**Pull request merged**